webapp/src/ee/billing/administration/subscriptionPlans/migration/cloud/AdministrationCloudPlanMigrationEdit.tsx

```diff
@@ -0,0 +1,187 @@
+  const { t } = useTranslate();
+  const match = useRouteMatch();
+  const messaging = useMessage();
+  const history = useHistory();
+  const formatDate = useDateFormatter();
+  const migrationId = match.params[PARAMS.PLAN_MIGRATION_ID] as number;
+  const [subscriptionsPage, setSubscriptionsPage] = useState(0);
+
+  const migrationLoadable = useBillingApiQuery({
+    url: '/v2/administration/billing/cloud-plans/migration/{migrationId}',
+    method: 'get',
+    path: { migrationId },
+  });
+
+  const subscriptions = useBillingApiQuery({
+    url: '/v2/administration/billing/cloud-plans/migration/{migrationId}/subscriptions',
+    method: 'get',
+    path: { migrationId },
+    query: {
+      page: subscriptionsPage,
+      size: 10,
+    },
+    options: {
+      keepPreviousData: true,
+    },
+  });
+
+  const updatePlanMigrationLoadable = useBillingApiMutation({
+    url: '/v2/administration/billing/cloud-plans/migration/{migrationId}',
+    method: 'put',
+  });
+
+  const deletePlanMigrationMutation = useBillingApiMutation({
+    url: '/v2/administration/billing/cloud-plans/migration/{migrationId}',
+    method: 'delete',
+  });
+
+  const onDelete = (migrationId: number) => {
+    deletePlanMigrationMutation.mutate(
+      { path: { migrationId } },
+      {
+        onSuccess: () => {
+          messaging.success(
+            <T keyName="administration_plan_migration_deleted_success" />
+          );
+          history.push(LINKS.ADMINISTRATION_BILLING_CLOUD_PLANS.build());
+        },
+      }
+    );
+  };
+
+  if (migrationLoadable.isLoading) {
+    return <SpinnerProgress />;
+  }
+
+  const migration = migrationLoadable.data!;
+
+  const submit = (values: PlanMigrationFormData) => {
+    updatePlanMigrationLoadable.mutate(
+      {
+        path: { migrationId },
+        content: { 'application/json': values },
+      },
+      {
+        onSuccess: () => {
+          messaging.success(
+            <T keyName="administration_plan_migration_updated_success" />
+          );
+          history.push(LINKS.ADMINISTRATION_BILLING_CLOUD_PLANS.build());
+        },
+      }
+    );
+  };
+
+  return (
+    <DashboardPage>
+      <BaseAdministrationView
+        windowTitle={t('administration_plan_migration_configure_existing')}
+        allCentered
+        hideChildrenOnLoading={false}
+        navigation={[
+          [
+            t('administration_cloud_plans'),
+            LINKS.ADMINISTRATION_BILLING_CLOUD_PLANS.build(),
+          ],
+          [
+            t('administration_plan_migration_configure_existing'),
+            LINKS.ADMINISTRATION_BILLING_CLOUD_PLAN_MIGRATION_EDIT.build({
+              [PARAMS.PLAN_MIGRATION_ID]: migrationId,
+            }),
+          ],
+        ]}
+      >
+        <Box>
+          <Typography variant="h5">
+            {t('administration_plan_migration_configure_existing')}
+          </Typography>
+        </Box>
+        <EditPlanMigrationForm
```

Comment:
This component is kind of long. I would extract this EditPlanMigrationForm to separate `CloudPlanEditPlanMigrationForm`, this would enable us to move also the hooks from top saving many lines from this component. Also submit and onDelete methods will be closer to their usage.